### DISCUSSION TITLE
Unexpected key(s) in state_dict when running Japanese TTS Training （日本語TTS学習でエラーが出ました）

### DISCUSSION THREAD

**Discussion by cryptowooser** [espnet/espnet#5239] [Q&A] [answered]
I am following along with the demo here:
https://qiita.com/RRR_troisR/items/6288b9bdc6e725aa8440
to get an ESPNET Japanese text to speech model working using VITS. The error below is unfortunately beyond my ability to debug (Though I see someone in [this thread](https://github.com/espnet/espnet/discussions/5211) seems to be having issues at the same stage as well) and it would be wonderful if someone could point me in the right direction. 

(Since the error involves Japanese TTS I will write the same message in Japanese)
このページに沿って日本語TTSモデルを学習させていますけど、最後のほうでエラーが出ます。
https://qiita.com/RRR_troisR/items/6288b9bdc6e725aa8440
自分の力では解決するのが難しそうなので、何か知っている人がいれば教えていただければ大変助かります。

I have completed all steps until the end when I run the following command: （最後のほうにある以降のコマンドを実行しますと）
$ ./run.sh \
    --stage 6 \
    --g2p pyopenjtalk_accent_with_pause \
    --min_wav_duration 0.38 \
    --fs 44100 \
    --n_fft 2048 \
    --n_shift 512 \
    --dumpdir dump/44k \
    --win_length null \
    --tts_task gan_tts \
    --feats_extract linear_spectrogram \
    --feats_normalize none \
    --train_config ./conf/tuning/finetune_full_band_vits.yaml \
    --train_args "--init_param downloads/c65dd99aa55a3c4fd6fcb15d3804e5cd/exp/tts_train_full_band_vits_raw_phn_jaconv_pyopenjtalk_accent_with_pause/train.total_count.ave_10best.pth:tts:tts" \
    --tag finetune_full_band_vits_raw_phn_jaconv_pyopenjtalk_accent_with_pause

Which raises the following error　（こんなふうなエラーが出ます）
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for VITS:
        Unexpected key(s) in state_dict: "discriminator.msd.discriminators.0.layers.0.0.weight", "discriminator.msd.discriminators.0.layers.1.0.weight", "discriminator.msd.discriminators.0.layers.2.0.weight", "discriminator.msd.discriminators.0.layers.3.0.weight", "discriminator.msd.discriminators.0.layers.4.0.weight", "discriminator.msd.discriminators.0.layers.5.0.weight", "discriminator.msd.discriminators.0.layers.6.0.weight", "discriminator.msd.discriminators.0.layers.7.weight".

**Comment by sw005320** [comment]
Thanks for the report, @cryptowooser.
Is it similar to https://github.com/espnet/espnet/issues/5237?

**Comment by cryptowooser** [comment]
I lack the fundamental understanding of how the system works to be 100% sure, but it would appear to be similar, yes!

**Comment by cryptowooser** [comment]
If it helps at all, I ran the code specified in 
https://github.com/espnet/espnet/issues/5237#issuecomment-1595660560
to generate a new .pth file. 
It runs now, but has very odd behavior: My GPU (A 4090) shoots to 23/24GB memory used, but GPU usage immediately goes from 100% to 0% utilization, while a single CPU core shoots to 100% utilization. Then it just kind of sits there.
It stayed like this for a full hour before I finally shut it down. I've verified that CUDA is installed on the machine and is available via torch. Below is the last line of the log: 

[DESKTOP-OE5OU32] 2023-06-19 14:13:13,322 (abs_task:1576) INFO: [plot_att] dataset:
ESPnetDataset(
  text: {"path": "dump/44k/raw/dev/text", "type": "text"}
  speech: {"path": "dump/44k/raw/dev/wav.scp", "type": "sound"}
  preprocess: <espnet2.train.preprocessor.CommonPreprocessor object at 0x7f7e00f322e0>)
[DESKTOP-OE5OU32] 2023-06-19 14:13:13,322 (abs_task:1577) INFO: [plot_att] Batch sampler: UnsortedBatchSampler(N-batch=20, batch_size=1, key_file=exp/tts_stats_raw_linear_spectrogram_phn_jaconv_pyopenjtalk_accent_with_pause/valid/text_shape.phn, 
[DESKTOP-OE5OU32] 2023-06-19 14:13:13,322 (abs_task:1578) INFO: [plot_att] mini-batch sizes summary: N-batch=3, mean=1.0, min=1, max=1
[DESKTOP-OE5OU32] 2023-06-19 14:13:13,426 (trainer:284) INFO: 1/100epoch started

**Reply by iamanigeeit** [reply]
Do you still have the same problem after the latest fix?

**Reply by cryptowooser** [reply]
I do, yes. I deleted the entire directory and redid the whole thing to make sure that wouldn't cause an issue, and the error hasn't changed.
I am currently using the master branch, and my current version is v.202304-442-gb5a88e954

The error is as follows. 
RuntimeError: Error(s) in loading state_dict for VITS:
        Unexpected key(s) in state_dict: "discriminator.msd.discriminators.0.layers.0.0.weight", "discriminator.msd.discriminators.0.layers.1.0.weight", "discriminator.msd.discriminators.0.layers.2.0.weight", "discriminator.msd.discriminators.0.layers.3.0.weight", "discriminator.msd.discriminators.0.layers.4.0.weight", "discriminator.msd.discriminators.0.layers.5.0.weight", "discriminator.msd.discriminators.0.layers.6.0.weight", "discriminator.msd.discriminators.0.layers.7.weight".
Accounting: time=5 threads=1
Ended (code 1) at Thu Jun 22 21:17:32 JST 2023, elapsed time 5 seconds

**Reply by kan-bayashi** [reply]
Hmm, let me reproduce.

**Reply by kan-bayashi** [reply]
I can reproduce and found the reason. Let me think solution.

**Reply by cryptowooser** [reply]
Thank you for checking into this so fast. I will await an update.

**Reply by kan-bayashi** [reply]
It should be fixed in #5249!

**Reply by cryptowooser** [reply]
It no longer gives me the error, but instead I'm getting the behavior described here:
https://github.com/espnet/espnet/discussions/5239#discussioncomment-6215373

CPU utilization switches to full utilization of 1 core while GPU utilization stays at 0, while GPU VRAM usage is at 23gigs/24gigs. 

This is using WSL, I have verified that torch is available and CUDA is installed.

**Reply by kan-bayashi** [reply]
Hmm, I’m not sure but did you check the CPU memory usage? When the process is silently died, sometimes CPU memory usage is full, i.e., OOM of CPU.

**Reply by cryptowooser** [reply]
It's currently using 18 gigs of memory, out of a 32 gig system.
I've tried reducing the number of batch jobs and it didn't work either. Can you reproduce on your end or is this just me? 
![image](https://github.com/espnet/espnet/assets/32764782/66ef6add-7df6-4fb3-865b-eb9ea1898f09)

**Reply by kan-bayashi** [reply]
I could not reproduce (I have no windows machine, sorry). I think WSL has the CPU memory allocation setting. Could you check it?

**Reply by cryptowooser** [reply]
I increased the memory to no avail, but I think it may be a WSL issue. I will try again shortly with a straight windows or maybe Linux build. Thanks for your help!

**Reply by cryptowooser** [reply]
The issue I was having was related to the swap file on my drive running out of space. It's since been resolved. 
Thanks again for your help!

**Comment by kan-bayashi** [comment, marked as answer]
Same as #4595 #5237 
Fixed in #5240 #5249
